x-pack/plugins/search_indices/public/hooks/api/use_index.ts

```diff
@@ -19,7 +19,7 @@ export const useIndex = (indexName: string) => {
     refetchInterval: POLLING_INTERVAL,
     refetchIntervalInBackground: true,
     refetchOnWindowFocus: 'always',
-    retry: 3,
+    retry: false,
```

Comment:
Discussed in the sync but we should fail immediately for 404 but for other errors, we can apply a retry logic. 

useQuery allows you to pass in a function and the second argument is the error object. I haven't tried this but you should be able to check the statusCode in error object and return false if its a 404 otherwise return true until failureCount === 3.

https://tanstack.com/query/latest/docs/framework/react/guides/query-retries

```Setting retry = (failureCount, error) => ... allows for custom logic based on why the request failed.
```

Reply:
updated in 4e07390eb41c6e74d3784e7bff1c677d719cf993